src/python/py/models/builder.py

```diff
@@ -21,16 +22,19 @@
 
 
 class Model:
-    def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
-        self.context_length = config.max_position_embeddings
-        self.original_context_length = config.original_max_position_embeddings if hasattr(config, "original_max_position_embeddings") else config.rope_scaling["original_max_position_embeddings"] if hasattr(config, "rope_scaling") and hasattr(config.rope_scaling, "original_max_position_embeddings") else config.max_position_embeddings
+    def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options): 
+        self.context_length = config.max_position_embeddings if hasattr(config, "max_position_embeddings") else config.seq_length
```

Comment:
I believe all the models previously supported by the builder include `max_position_embeddings` in their configuration. However, since `chatglm3` does not have this parameter, we introduced `seq_length` to ensure the configuration remains compatible.

Reply:
It appears that `chatglm3` has the `seq_length` parameter [here](https://huggingface.co/THUDM/chatglm3-6b/blob/91a0561caa089280e94bf26a9fc3530482f0fe60/config.json#L35). Since most models use `config.max_position_embeddings` instead of `config.seq_length`, can we rewrite this to the following?

```python
self.context_length = config.seq_length if hasattr(config, "seq_length") else config.max_position_embeddings
```

Reply:
I agree. This is a better way to write this config. Incorporated in config `refactor_context_length_config` .
Thanks!